src/wms/src/main/java/org/geoserver/wms/WMS.java

```diff
@@ -1274,6 +1274,33 @@ public boolean isQueryable(LayerGroupInfo layerGroup) {
         return dimensions.getTimeDomain(queryRange, maxAnimationSteps);
     }
 
+    /**
+     * Query and returns the requested times for the given layer
+     *
+     * @throws IOException
+     */
+    public TreeSet<Date> queryCoverageNearestMatchTimes(
+            CoverageInfo coverage, List<Object> queryRanges) throws IOException {
+        DimensionInfo time = coverage.getMetadata().get(ResourceInfo.TIME, DimensionInfo.class);
+        final TreeSet<Date> foundDates = new TreeSet<Date>();
+        if (time == null || !time.isEnabled()) {
+            throw new ServiceException(
+                    "Coverage " + coverage.prefixedName() + " does not have time support enabled");
+        }
+
+        List<Object> dates = getNearestMatches(coverage, time, queryRanges, ResourceInfo.TIME);
```

Comment:
Same comments as in the enum above... come to think of it, the code is pretty similar, is there duplication here?

Reply:
Actually not. The available WMS.queryCoverageTimes was getting one dateRange at time so the "aggregation" is made outside by the enum method doing the query calls.
The new WMS.queryNearestCoverageTimes accept multiple times (a List) so the aggregation is made inside.
I'm fixing the sequential + containment checks here too.